code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -20,8 +20,43 @@
 	skin_material = /decl/material/solid/skin/goat
 	skin_amount = 8
 
+	ai = /datum/ai/goat
+
 	var/datum/reagents/udder = null
 
+/datum/ai/goat
+	expected_type = /mob/living/simple_animal/hostile/retaliate/goat
+
+/datum/ai/goat/do_process(time_elapsed)
+	. = ..()
+	var/mob/living/simple_animal/hostile/retaliate/goat/goat = body
+
+	//chance to go crazy and start wacking stuff
+	if(!length(goat.enemies) && prob(1))
+		goat.Retaliate()
+
+	if(length(goat.enemies) && prob(10))
+		goat.enemies = list()
+		goat.LoseTarget()
+		goat.visible_message(SPAN_NOTICE("\The [goat] calms down."))
+
+	var/obj/effect/vine/SV = locate() in goat.loc
+	if(SV && prob(60))
+		goat.visible_message(SPAN_NOTICE("\The [goat] eats the plants."))
+		SV.die_off(1)
+		var/obj/machinery/portable_atmospherics/hydroponics/soil/invisible/SP = locate() in goat.loc
+		if(SP)
+			qdel(SP)
+		else if(prob(20))
```

Comment:
This is technically a functional change, from `!prob(60) && prob(20)` to `prob(60) && prob(20)`. I don't particularly mind it though.